### PR TITLE
build-configs: Enable coverage of 64K pages on arm64

### PR DESCRIPTION
**Pull request by broonie**
arm64 supports a range of page sizes, including 64K. In testing on other
systems we seem to turn up a small but interesting stream of bugs on 64K
pages which it'd be good to keep an eye on so let's enable a build with
that config option enabled.

This is added for clang-9 simply because we're not really focused on which
compiler is used for the coverage but clang-9 is the only compiler which
has a list of configs already specified so it's the least invasive change.

Signed-off-by: Mark Brown <broonie@kernel.org>

### CODE REVIEW

**Comment by gctucker** [comment]
Maybe also worth mentioning that this only enables it on linux-next/master.

**Comment by gctucker** [comment]
Testing on staging now with a one-off build of linux-next.

**Comment by gctucker** [comment]
Well we're back to this issue with `merge_config.sh` failing with Clang:
https://staging.kernelci.org/build/id/5f36cb8b6cb48bb582d48da6/

@nickdesaulniers What's the status of that getting fixed upstream? I thought we could now use a new environment variable to switch everything to Clang and propagate that through `merge_config.sh`.  There are several issues open around that, I've kind of lost track of the current status.

**Comment by nickdesaulniers** [comment]
I just found out @adelva1984 is using merge_config.sh within Android, but I thought that was broken? (I forget the details).

**Comment by gctucker** [comment]
@broonie Merge this to remove your YAML pain: https://github.com/broonie/kernelci-core/pull/2

**Comment by gctucker** [comment]
Tested OK with 2 extra builds on staging with this revision:
https://staging.kernelci.org/build/kernelci/branch/staging.kernelci.org/kernel/staging-20200901.3/

**Comment by gctucker** [comment]
Test results can be seen here:
https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200901.3/plan/baseline/

**Comment by gctucker** [comment]
However, it looks like the 64K config is not actually enabled as it ought to be:
https://storage.staging.kernelci.org/kernelci/staging.kernelci.org/staging-20200901.3/arm64/defconfig+CONFIG_ARM64_64K_PAGES/clang-10/kernel.config
```
CONFIG_ARM64_4K_PAGES=y
# CONFIG_ARM64_16K_PAGES is not set
# CONFIG_ARM64_64K_PAGES is not set
# CONFIG_ARM64_VA_BITS_39 is not set
CONFIG_ARM64_VA_BITS_48=y
CONFIG_ARM64_VA_BITS=48
CONFIG_ARM64_PA_BITS_48=y
CONFIG_ARM64_PA_BITS=48
# CONFIG_CPU_BIG_ENDIAN is not set
CONFIG_CPU_LITTLE_ENDIAN=y
```

**Comment by gctucker** [comment]
Same thing with the gcc-8 build:
https://storage.staging.kernelci.org/kernelci/staging.kernelci.org/staging-20200901.3/arm64/defconfig+CONFIG_ARM64_64K_PAGES/gcc-8/kernel.config
```
CONFIG_ARM64_4K_PAGES=y
# CONFIG_ARM64_16K_PAGES is not set
# CONFIG_ARM64_64K_PAGES is not set
# CONFIG_ARM64_VA_BITS_39 is not set
CONFIG_ARM64_VA_BITS_48=y
CONFIG_ARM64_VA_BITS=48
CONFIG_ARM64_PA_BITS_48=y
CONFIG_ARM64_PA_BITS=48
# CONFIG_CPU_BIG_ENDIAN is not set
CONFIG_CPU_LITTLE_ENDIAN=y
```

**Comment by gctucker** [comment]
I believe it's because the config option is missing `=y`:
https://storage.staging.kernelci.org/kernelci/staging.kernelci.org/staging-20200901.3/arm64/defconfig+CONFIG_ARM64_64K_PAGES/clang-10/frag.config
```
CONFIG_ARM64_64K_PAGES
```

File: build-configs.yaml [hunk 1]
```diff
@@ -680,7 +680,14 @@ build_configs:
           mips: *mips_arch
           riscv: *riscv_arch
           arc: *arc_arch
-          arm64: *arm64_arch
+          arm64:
+            <<: *arm64_arch
+            extra_configs:
+              - 'allmodconfig'
+              - 'allnoconfig'
+              - 'defconfig+CONFIG_ARM64_64K_PAGES'
```
Comment (gctucker) [inline build-configs.yaml:688]:
```suggestion
              - 'defconfig+CONFIG_ARM64_64K_PAGES=y'
```

File: build-configs.yaml [hunk 2]
```diff
@@ -697,6 +704,7 @@ build_configs:
             extra_configs:
               - 'allmodconfig'
               - 'allnoconfig'
+              - 'defconfig+CONFIG_ARM64_64K_PAGES'
```
Comment (gctucker) [inline build-configs.yaml:707]:
```suggestion
              - 'defconfig+CONFIG_ARM64_64K_PAGES=y'
```

**Review by gctucker — COMMENTED** [review]
@broonie Please see above, I think there was a slight oversight with the `=y` thing.

**Review by gctucker — APPROVED** [review]
Tested OK on staging, build and runtime:
https://staging.kernelci.org/test/plan/id/5f51fb0c1f33ad5c20d65a1d/

**Pull request merged**